code/parse/sexp.cpp

```diff
@@ -1947,6 +1948,29 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 		} else if (Sexp_nodes[node].subtype == SEXP_ATOM_STRING) {
 			type2 = SEXP_ATOM_STRING;
 
+		} else if (Sexp_nodes[node].subtype == SEXP_ATOM_CONTAINER) {
+			const auto *p_container = get_sexp_container(Sexp_nodes[node].text);
+			Assert(p_container != nullptr); // should have been checked in get_sexp()
+
+			switch (type) {
+				case OPF_CONTAINER_NAME:
+				case OPF_LIST_CONTAINER_NAME:
+				case OPF_MAP_CONTAINER_NAME:
+					// set two values for extra certainty
+					// since SEXP_ATOM_CONTAINER == OPR_AI_GOAL (yay untyped enums)
```

Comment:
It shouldn't matter that these enums overlap.  Where is the other section of code that motivated the "set two values" redundancy here?

Reply:
I removed this code as part of an overhaul of SEXP syntax checking for containers.